[CI] need to resolve issues reported by shellcheck

##### SUMMARY
The following CI tests have been failed with shellcheck error:

- ansible-test-sanity-docker-devel
- ansible-test-sanity-docker-milestone

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
-  tests/utils/shippable/shippable.sh

##### ANSIBLE VERSION
```
- devel
- milestone
```

##### COLLECTION VERSION
```
- main
```

##### CONFIGURATION
None

##### OS / ENVIRONMENT
None

##### STEPS TO REPRODUCE
- Create a pull request

##### EXPECTED RESULTS
The following sanity tests should be passed without errors:

- ansible-test-sanity-docker-devel
- ansible-test-sanity-docker-milestone

##### ACTUAL RESULTS
CI tests failed with the following error:

```
2021-12-19 06:06:53.946292 | fedora-35 | Run command: /root/.ansible/test/venv/sanity/3.8/4f53cda1/bin/python /root/ansible/test/lib/ansible_test/_util/controller/sanity/code-smell/shebang.py
2021-12-19 06:06:54.037158 | fedora-35 | Running sanity test "shellcheck"
2021-12-19 06:06:54.043132 | fedora-35 | Run command: shellcheck -e SC1090,SC1091,SC2164 --format checkstyle .azure-pipelines/scripts/aggregate-coverage.sh .azure-pipelines/scripts/process-results.sh .azure-pipelines/scripts/publish-codecov.sh .azure-pipelines/scripts/report-coverage.sh .azure-pipelines/scripts/run-tests.sh tests/utils/shippable/cloud.sh tests/utils/shippable/linux.sh tests/utils/shippable/remote.sh tests/utils/shippable/sanity.sh tests/utils/shippable/shippable.sh tests/utils/shippable/timing.sh tests/utils/shippable/units.sh
2021-12-19 06:06:55.522440 | fedora-35 | ERROR: Found 1 shellcheck issue(s) which need to be resolved:
2021-12-19 06:06:55.523564 | fedora-35 | ERROR: tests/utils/shippable/shippable.sh:53:10: SC2242: Can only exit with status 0-255. Other data should be written to stdout/stderr.
2021-12-19 06:06:55.526222 | fedora-35 | See documentation for help: https://docs.ansible.com/ansible-core/devel/dev_guide/testing/sanity/shellcheck.html
```

--- a/changelogs/fragments/302_shippable_exit_code.yml
+++ b/changelogs/fragments/302_shippable_exit_code.yml
@@ -1,0 +1,3 @@
+---
+trivial:
+- CI tests - fix exit code to address shellckeck test issue (https://github.com/ansible-collections/ansible.posix/issues/301).

--- a/tests/utils/shippable/shippable.sh
+++ b/tests/utils/shippable/shippable.sh
@@ -50,7 +50,7 @@ function retry
         echo "@* -> ${result}"
     done
     echo "Command '@*' failed 3 times!"
-    exit -1
+    exit 255
 }
 
 command -v pip

